File: AdvantageFramework.Core/Database/Entities/DocumentType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("DOCUMENT_TYPE")]
    public partial class DocumentType
    {
        public DocumentType()
        {
            Documents = new HashSet<Document>();
        }

        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        [Key]
        [Column("DOCUMENT_TYPE_ID")]
        public int DocumentTypeId { get; set; }
        [Column("DOCUMENT_TYPE_DESC")]
        [StringLength(20)]
        public string DocumentTypeDesc { get; set; }
        [Column("INACTIVE_FLAG")]
        public bool? InactiveFlag { get; set; }
        [Column("IS_DEFAULT")]
        public bool? IsDefault { get; set; }

        [InverseProperty(nameof(Document.DocumentType))]
        public virtual ICollection<Document> Documents { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/DocumentType.vb
Namespace Database.Entities

	<Table("DOCUMENT_TYPE")>
	Public Class DocumentType
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

		Public Enum Properties
			ID
			Description
			Documents

		End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

		<Key>
		<Required>
		<Column("DOCUMENT_TYPE_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property ID() As Integer
		<MaxLength(20)>
		<Column("DOCUMENT_TYPE_DESC", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Description() As String

        Public Overridable Property Documents As ICollection(Of Database.Entities.Document)

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID.ToString

        End Function

#End Region

	End Class

End Namespace
